test/foundry/NewMCR.t.sol

```diff
@@ -0,0 +1,239 @@
+        vm.assume(amount2 > 0 && amount2 < 2500000000000);
+        vm.assume(amount3 > 0 && amount3 < 2500000000000);
+        vm.assume(amount4 > 0 && amount4 < 2500000000000);
+
+        // Mint
+        vm.prank(MintAddress);
+        usdc.mint(address(user), 5000000000000);
+        vm.prank(MintAddress);
+        usdc.mint(address(user2), amount);
+        vm.prank(MintAddress);
+        usdc.mint(address(user3), amount2);
+        vm.prank(MintAddress);
+        usdc.mint(address(user4), amount);
+        vm.prank(MintAddress);
+        usdc.mint(address(user5), amount2);
+        vm.prank(MintAddress);
+        usdc.mint(address(user6), amount4);
+        vm.prank(MintAddress);
+        usdc.mint(address(user7), amount);
+        vm.prank(MintAddress);
+        usdc.mint(address(user8), amount3);
+
+        // Approve
+        vm.prank(address(user));
+        usdc.approve(address(pool), 5000000000000);
+        vm.prank(address(user2));
+        usdc.approve(address(pool), amount);
+        vm.prank(address(user3));
+        usdc.approve(address(pool), amount2);
+        vm.prank(address(user4));
+        usdc.approve(address(pool), amount);
+        vm.prank(address(user5));
+        usdc.approve(address(pool), amount2);
+        vm.prank(address(user6));
+        usdc.approve(address(pool), amount4);
+        vm.prank(address(user7));
+        usdc.approve(address(pool), amount);
+        vm.prank(address(user8));
+        usdc.approve(address(pool), amount3);
+
+        // Enter in pool
+        vm.prank(address(user));
+        pool.enterInPool(500000000000);
+        vm.prank(address(user2));
+        pool.enterInPool(amount);
+        vm.prank(address(user3));
+        pool.enterInPool(amount2);
+        vm.prank(address(user4));
+        pool.enterInPool(amount);
+        vm.prank(address(user5));
+        pool.enterInPool(amount2);
+        vm.prank(address(user6));
+        pool.enterInPool(amount4);
+        vm.prank(address(user7));
+        pool.enterInPool(amount);
+        vm.prank(address(user8));
+        pool.enterInPool(amount3);
+    }
+
+    function test_initialSetupAndStaking(uint256 amount, uint256 amount2, uint256 amount3, uint256 amount4) public {
+        setupUsersAndStakes(amount, amount2, amount3, amount4);
+
+        uint256 expectedStake = 500000000000 + (3 * amount) + (2 * amount2) + amount3 + amount4;
+        assertEq(proxycapital.totalCapitalStaked(), expectedStake);
+    }
+
+    function test_settingMCR(uint256 amount, uint256 amount2, uint256 amount3, uint256 amount4) public {
+        setupUsersAndStakes(amount, amount2, amount3, amount4);
+
+        uint256 expectedStake = 2500000000000 + (3 * amount) + (2 * amount2) + amount3 + amount4;
+        proxycapital.setMCR(expectedStake / 2);
+
+        assertEq(proxycapital.MCR(), (expectedStake / 2));
+    }
+
+    function test_withdrawalProcess(uint256 amount, uint256 amount2, uint256 amount3, uint256 amount4) public {
```

Comment:
You can also write a test for the MCR where :
1. `user` can withdraw all their money in the beginning of the test
2. `user` stakes again all his money 
3.  others users withdraw their money
4. `user` can't withdraw everything anymore because there is less money in the pool so he needs to leave some money there